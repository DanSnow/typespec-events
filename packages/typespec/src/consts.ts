export const PACKAGE_NAME = '@typespec-events/typespec';
