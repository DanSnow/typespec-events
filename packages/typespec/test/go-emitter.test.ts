import { describe, expect, it } from 'vitest';
import { PACKAGE_NAME } from '../src/consts.js';
import { emit } from './test-host.js';

describe('go emitter', () => {
  it('emit events.go with camelCase (default)', async () => {
    const results = await emit('@event("cta_clicked") model CtaClicked { title: string }', {
      options: {
        [PACKAGE_NAME]: {
          languages: ['go'],
        },
      },
    });
    expect(results['events.go']).toMatchSnapshot();
  });

  it('emit events.go with PascalCase', async () => {
    const results = await emit('@event("cta_clicked") model CtaClicked { title: string }', {
      options: {
        [PACKAGE_NAME]: {
          languages: ['go'],
          schemaNamingConvention: 'PascalCase',
        },
      },
    });
    expect(results['events.go']).toMatchSnapshot();
  });

  it('emit nested models in correct order', async () => {
    const results = await emit(
      `
      model City { name: string, zip: int32 }
      model Address { street: string, city: City }
      @event("user_address_updated")
      model UserAddressUpdated { userId: string, address: Address }
    `,
      {
        options: {
          [PACKAGE_NAME]: {
            languages: ['go'],
          },
        },
      },
    );
    expect(results['events.go']).toMatchSnapshot();
  });
});
