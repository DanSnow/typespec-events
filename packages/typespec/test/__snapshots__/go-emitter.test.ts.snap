// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`go emitter > emit events.go with PascalCase 1`] = `
"package main


type CtaClicked struct { Title string \`json:"title"\` }


// Event Schemas
var EventSchemas = map[string]interface{}{
  "cta_clicked": CtaClicked,
}"
`;

exports[`go emitter > emit events.go with camelCase (default) 1`] = `
"package main


type CtaClicked struct { Title string \`json:"title"\` }


// Event Schemas
var EventSchemas = map[string]interface{}{
  "cta_clicked": CtaClicked,
}"
`;

exports[`go emitter > emit nested models in correct order 1`] = `
"package main


type City struct { Name string \`json:"name"\`; Zip int \`json:"zip"\` }

type Address struct { Street string \`json:"street"\`; City City \`json:"city"\` }

type UserAddressUpdated struct { UserId string \`json:"userId"\`; Address Address \`json:"address"\` }


// Event Schemas
var EventSchemas = map[string]interface{}{
  "user_address_updated": UserAddressUpdated,
}"
`;
